compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -126,10 +125,57 @@ where
         }
     }
 
+    let infcx_compat = infcx.fork();
+
+    debug!(?assumed_wf_types);
+    let implied_bounds = infcx.implied_bounds_tys(param_env, body_def_id, &assumed_wf_types);
     let outlives_env = OutlivesEnvironment::with_bounds(param_env, implied_bounds);
 
-    wfcx.ocx.resolve_regions_and_report_errors(body_def_id, &outlives_env)?;
-    infcx.tainted_by_errors().error_reported()
+    let errors = infcx.resolve_regions(&outlives_env);
+    if errors.is_empty() {
+        return Ok(());
+    }
+
+    let is_bevy = 'is_bevy: {
+        // We don't want to emit this for dependents of Bevy, for now.
```

Comment:
What tracking issue? There isn't one currently open. Do you want me to open specifically to track this special-case?

Reply:
yeah, I think it would be valuable to open a tracking issue for this special-case. Otherwise we may not fix it for bevy (or get surprised issues when bevy breaks because they end up renaming their types :sweat_smile:)

Reply:
Opened #119956 and linked.